src/object.rs

```diff
@@ -414,4 +419,60 @@ impl<T: IsA<Object>> ObjectExt for T {
             (Some(ptype), Some(type_)) => ptype == type_,
         }
     }
+
+    fn connect<'a, N, F>(&self, signal_name: N, after: bool, callback: F) -> Result<u64, BoolError>
+        where N: Into<&'a str>, F: Fn(&[Value]) -> Option<Value> + Send + Sync + 'static {
+        let signal_name: &str = signal_name.into();
+
+        unsafe {
+            let obj = self.to_glib_none().0;
+            let klass = (*obj).g_type_instance.g_class as *mut gobject_ffi::GTypeClass;
+            let type_ = (*klass).g_type;
+
+            let mut signal_id = 0;
+            let mut signal_detail = 0;
+
+            let found: bool = from_glib(gobject_ffi::g_signal_parse_name(signal_name.to_glib_none().0,
+                                                                         type_, &mut signal_id,
+                                                                         &mut signal_detail, true.to_glib()));
+
+            if !found {
+                return Err(BoolError("Signal not found"));
+            }
+
+            let mut details = mem::zeroed();
+            gobject_ffi::g_signal_query(signal_id, &mut details);
+            if details.signal_id != signal_id {
+                return Err(BoolError("Signal not found"));
+            }
+
+            // This is actually G_SIGNAL_TYPE_STATIC_SCOPE
+            let return_type = details.return_type & (!gobject_ffi::G_TYPE_FLAG_RESERVED_ID_BIT);
+            let closure = Closure::new(move |values| {
+                let ret = callback(values);
+
+                if return_type == gobject_ffi::G_TYPE_NONE {
+                    // Silently drop return value
+                    None
+                } else {
+                    // Silently create empty return value
+                    if let None = ret {
+                        let mut value = Value::uninitialized();
+                        gobject_ffi::g_value_init(value.to_glib_none_mut().0, return_type);
+                        Some(value)
+                    } else {
+                        ret
```

Comment:
Also in this case we want to check if ret's type_ == return_type too, and if not do the same as if None was returned